### DISCUSSION TITLE
Integrate Updates with Winget Community

### DISCUSSION THREAD

**Discussion by SirLouen** [ente-io/ente#3400] [Enhancements]
The idea is to create msi instead of exe packages from this repository and add them to winget Community Repository.
https://github.com/microsoft/winget-pkgs

Instead of using Inno Setup for packaging, consider something like MSIX https://pub.dev/packages/msix

https://learn.microsoft.com/en-us/windows/package-manager/package/
Create the manifest: https://learn.microsoft.com/en-us/windows/package-manager/package/manifest?tabs=minschema%2Cversion-example
Submit the manifest: https://learn.microsoft.com/en-us/windows/package-manager/package/repository

**Comment by jo-chemla** [comment]
Why build msi when winget-pkgs also accept standalone exe files?

**Reply by SirLouen** [reply]
> Why build msi when winget-pkgs also accept standalone exe files?

I was reading winget docs, but now i also see it accepts standalone exe, so better.

**Reply by jo-chemla** [reply]
The package manifest for [ente-auth](https://github.com/microsoft/winget-pkgs/blob/master/manifests/e/ente-io/auth-desktop/4.0.2/ente-io.auth-desktop.installer.yaml) installer already lists `InstallerUrl: https://github.com/ente-io/ente/releases/download/auth-v4.0.2/ente-auth-v4.0.2-installer.exe`

**Reply by SirLouen** [reply]
I can see it's already there https://winstall.app/apps/ente-io.auth-desktop

For some reason when I posted this I could not find it.

**Reply by jo-chemla** [reply]
Great, probably the package was no there yet (first auth was 5 months ago) or winstall db was not up-to-date. You can probably close the discussion then!

**Reply by SirLouen** [reply]
Something is going wrong with the package

```
> winget update --id=ente-io.auth-desktop  -e
No available upgrade found.
No newer package versions are available from the configured sources.
```

![image](https://github.com/user-attachments/assets/96896c35-073d-466b-b5a1-aa1cfc7b2f95)

**Reply by jo-chemla** [reply]
Latest release of `ente-io.auth-desktop` on winget-pkgs is `4.0.2` see https://github.com/microsoft/winget-pkgs/tree/master/manifests/e/ente-io/auth-desktop
If you want ente to push upgrades to winget-pkgs on every github release, please track the following thread with suggested github-action workflow based on winget-releaser: https://github.com/ente-io/ente/discussions/3950
